pylsp_jsonrpc/endpoint.py

```diff
@@ -35,9 +37,24 @@ def __init__(self, dispatcher, consumer, id_generator=lambda: str(uuid.uuid4()),
         self._client_request_futures = {}
         self._server_request_futures = {}
         self._executor_service = futures.ThreadPoolExecutor(max_workers=max_workers)
+        self._cancelledRequests = set()
+        self._messageQueue = None
+        self._consume_task = None
+
+    def init_async(self):
+        self._messageQueue = asyncio.Queue()
+        self._consume_task = asyncio.create_task(self.consume_task())
```

Comment:
This task will not run in an executor from https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor, but needs to be started after the event loop was started. I pushed an example implementation to https://github.com/davschul/python-lsp-server/commit/9f0f03d27e40ec8cb16a78195747f64c29129298

Reply:
The infinite loop has potential to block the main event loop, which could lead to performance degradation/unexpected blockage. I think in this case it would be better to run this on a separate event loop

Reply:
The asyncio queues used to send the messages cannot be used from different event loops. What might cause the block here in your opinion?

Reply:
Sorry for the long break, I was side tracked a bit in the python lsp support in Qt Creator and some urgent bugs before our release. I added a restriction to the loop, but I guess that not really what you've expected right?